### ISSUE TITLE
au failing to push package updates with PowerShell Core 7.3

### ISSUE THREAD

**Issue by flcdrg** [chocolatey-community/chocolatey-au#24]
There seems to be a breaking change between PS 7.2 and 7.3 that is causing AU to fail when trying to push new package versions.

I can see Appveyor seems to have upgraded their images from 7.2 to 7.3 a few days ago and since then all my updates are failing. eg.

>    [54/55] sqltoolbelt ERROR: 
     Chocolatey v1.3.0 File specified is either not found or not a .nupkg file. 'sqltoolbelt.2023.03.01.nupkg ' (50.30s)
   [55/55] vt-cli ERROR: 
     Chocolatey v1.3.0 File specified is either not found or not a .nupkg file. 'vt-cli.0.12.0.nupkg ' (19.76s)

The extra trailing space does look suspicious

**Comment by flcdrg** [comment]
[Previously mentioned in comment](https://github.com/chocolatey-community/chocolatey-au/issues/13#issuecomment-1447954701)

**Comment by flcdrg** [comment]
Spent a few hours trying to figure out what's going on, and I'm stumped. Not easy to debug as the codebase uses PowerShell Jobs, and as far as I can tell, you can't set breakpoints/step through Jobs. Also weird, I get different behaviour running in VS Code vs from PowerShell in Terminal. Very weird.

**Comment by tunisiano187** [comment]
Same for me

**Comment by flcdrg** [comment]
FWIW I have a branch on my fork that is working - https://github.com/flcdrg/chocolatey-au/tree/fix-warnings

**Comment by flcdrg** [comment]
And now my branch stopped working (mostly because I think GitHub Actions upgraded the version of PowerShell they use)

**Comment by flcdrg** [comment]
My workaround was to rewrite the "orchestration" part of AU in C#/.NET - https://github.com/flcdrg/au-dotnet/

I'm still using it for the actual `update.ps1` stuff though - the .NET code hosts PowerShell and calls into that)

It works pretty well for me and fixes the problem with weird spurious additional spaces appearing where they're not wanted.

**Comment by stablestud** [comment]
Unfortunately it is also broken for me, pushing from Github Actions fails due to this strange trailing whitespace.

@flcdrg I tried to use your `au-dotnet` but I fail to use it as I have no idea how to use it. I am not a C#/Windows developer therefore a nice tutorial would be nice how to set it up. Would be nice if it could just be installed via `choco install`

**Comment by flcdrg** [comment]
Yeah, It's something I 'cobbled' together but you're right. I need to document it and make it easier to use

**Comment by stablestud** [comment]
Using `powershell` instead of `pwsh` as shell in Github Actions seems to have fixed the issue.
Using `powershell` will use Powershell 5.1 instead of 7.x

@tunisiano187 this "fix" might be interesting for you

**Comment by flcdrg** [comment]
That is true, but then you're limited to using PowerShell 5 for your update.ps1 scripts (which in my case was a blocker)

**Comment by flcdrg** [comment]
@stablestud I've improved the docs for au-dotnet and created a NuGet package for it so it is much easier to consume.

Take a look at my updated workflow - https://github.com/flcdrg/au-packages/blob/master/.github/workflows/ci.yml#L28 which I'm using in my packages repo.
